src/AtelierTomato.Markov.Service.Discord/DiscordSentenceBuilder.cs

```diff
@@ -0,0 +1,23 @@
+﻿using AtelierTomato.Markov.Model;
+using AtelierTomato.Markov.Model.ObjectOID;
+using Discord;
+
+namespace AtelierTomato.Markov.Service.Discord
+{
+	public class DiscordSentenceBuilder
+	{
+		public static async Task<IEnumerable<Sentence>> Build(IGuild? guild, IChannel channel, ulong messageID, ulong userID, DateTimeOffset date, IEnumerable<string> sentenceTexts, string instance = "discord.com")
+		{
+			DiscordObjectOID OID = await DiscordObjectOIDBuilder.Build(guild, channel, messageID, instance);
+			AuthorOID author = new(ServiceType.Discord, instance, userID.ToString());
+			return sentenceTexts.Select((text, index) =>
+				new Sentence(
+					DiscordObjectOID.ForSentence(OID.Instance, OID.Server!.Value, OID.Category!.Value, OID.Channel!.Value, OID.Thread!.Value, OID.Message!.Value, index),

```

Comment:
There has to be a better way than to copy with mass exclamation mark / a ton of null checks ...